packages/shared/src/hooks/utils/useDownloadUrl.ts

```diff
@@ -0,0 +1,35 @@
+interface DownloadProps {
+  filename: string;
+  url: string;
+}
+
+interface UseDownloadUrl {
+  onDownloadUrl: (props: DownloadProps) => Promise<void>;
+}
+
+interface UseDownloadUrlProps {
+  onError?: (error: Error) => void;
+}
+
+export const useDownloadUrl = ({
+  onError,
```

Comment:
I would move this to `onDownloadUrl` function call instead since this is where handling of error actually happens.

Reply:
Thinking about it more, it might even be better to just do the handling with try catch in place you call `onDownloadUrl`. Since onError is optional errors would be swallowed like this which is bad pattern.

Reply:
Yeah, that is why I think this actually might be better https://github.com/dailydotdev/apps/pull/2603#discussion_r1486076862

Reply:
Yep. Let's go for that.